template/Makefile.j2

```diff
@@ -0,0 +1,50 @@
+# =============
+# This file is automatically generated from the templates in stackabletech/operator-templating
+# DON'T MANUALLY EDIT THIS FILE
+# =============
+
+.PHONY: docker chart-lint compile-chart
+
+TAG    := $(shell git rev-parse --short HEAD)
+
+VERSION := $(cargo metadata --format-version 1 | jq '.packages[] | select(.name=="stackable-{[ operator.product_string }]-operator") | .version')
+
+docker:
+	docker build --force-rm -t "docker.stackable.tech/stackable/{[ operator.product_string }]-operator:${TAG}" -t "docker.stackable.tech/stackable/{[ operator.product_string }]-operator:latest" -f docker/Dockerfile .
+	echo "${NEXUS_PASSWORD}" | docker login --username github --password-stdin docker.stackable.tech
+	docker push --all-tags docker.stackable.tech/stackable/{[ operator.product_string }]-operator
+
+## Chart related targets
+compile-chart: version crds config
+
+chart-clean:
+	rm -rf deploy/helm/{[ operator.product_string }]-operator/configs
+	rm -rf deploy/helm/{[ operator.product_string }]-operator/templates/crds.yaml
+
+version:
+	sed -i deploy/helm/{[ operator.product_string }]-operator/Chart.yaml "<version_placeholder>" ${VERSION}
+
+config: deploy/helm/{[ operator.product_string }]-operator/configs
+
+deploy/helm/{[ operator.product_string }]-operator/configs:
+	cp -r deploy/config-spec deploy/helm/{[ operator.product_string }]-operator/configs
+
+crds: deploy/helm/{[ operator.product_string }]-operator/templates/crds.yaml
```

Comment:
CRDs should generally be in `$CHART_HOME/crds/`, which ensures that they are installed before the chart itself, and that they are not deleted even if the operator chart is.

Reply:
If you place CRDS in the `$CHART_HOME/crds/` directory, helm will not update them. You have to update them manually. To prevent the deletion on helm chart uninstall, I added the helm hook: `resource-policy=keep`:
https://github.com/stackabletech/operator-templating/pull/7/files#diff-9e6f51403e0a072f731bafd87e9c7d4ed0dca19e57cd445df0aaa31dc7d1dac8R36

Reply:
The operator itself does not need the charts before you can install it.

Reply:
https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations

Reply:
> The operator itself does not need the charts before you can install it.

That's true, but it prevents the same Helm Release from defining an object of that resource. That becomes an issue if any users try to pull in our operator as a Helm dependency.

> To prevent the deletion on helm chart uninstall, I added the helm hook: resource-policy=keep:

That's just delaying the issue though, since that means that reinstalling the chart will fail with a conflict error.

Reply:
So we have to decide. Either the helm chart can not update CRDs (putting them in the `crds` directory), or a user can't specify the helm-operator chart as depedency, if they install stackable resources in their helm chart. I think the first usecase (update CRDs) is much more common than the second usecase.